cicecore/cicedynB/infrastructure/ice_grid.F90

```diff
@@ -59,19 +65,29 @@ module ice_grid
          dyt    , & ! height of T-cell through the middle (m)
          dxu    , & ! width of U-cell through the middle (m)
          dyu    , & ! height of U-cell through the middle (m)
+         dxn    , & ! width of N-cell through the middle (m)
+         dyn    , & ! height of N-cell through the middle (m)
+         dxe    , & ! width of E-cell through the middle (m)
```

Comment:
CICE only reads HTN and HTE.  All other grid lengths are derived from those.  The current extension to N and E grids is done the "same way".  That method is far from ideal (does not conserve area locally), but it's what we've had for a long time.  I think it's worthwhile looking into whether we can improve the solution at some point down the road.

Reply:
I'll admit I'm being a little picky here, but here's what's available from the grid file:

> [eclare@ba-fe1 gx3]$ ncdump -h grid_gx3.nc
> netcdf grid_gx3 {
> dimensions:
>         y = 116 ;
>         x = 100 ;
> variables:
>         double ulat(y, x) ;
>                 ulat:units = "rad" ;
>         double ulon(y, x) ;
>                 ulon:units = "rad" ;
>         double htn(y, x) ;
>                 htn:units = "cm" ;
>         double hte(y, x) ;
>                 hte:units = "cm" ;
>         double hus(y, x) ;
>                 hus:units = "cm" ;
>         double huw(y, x) ;
>                 huw:units = "cm" ;
>         double angle(y, x) ;
>                 angle:units = "rad" ;

It would be prudent to read in hus and huw, and then set dxe = hus and dyn = huw, at least for now.  How different are the resulting grid lengths, from using the averaging method?  Hopefully not very much...  but does that help the area conservation issue?  Using existing grid file variables would save some pretty substantial chunks of code in ice_grid.F90, for calculating dxe and dyn.

Reply:
We're opening a huge can of worms here.  Using HUS and HUW will almost certainly improve the accuracy of the grid and local area conservation.  But, if we read them, then dxt and dyt should also probably change.  Yes, dxe=HUS.  But then dxt should be 0.5*(HUSij + HUSi-1j) and that will change all answers, make CICE inconsistent with POP (and impact POP/CICE coupling conservation?), and possibly break some usage if users have grid files that don't include HUS and HUW.  I'm willing to go there, but we should also carefully examine the solutions that are produced.  We could run some qc as a starting point.  Do we want to do this now or is that a separate task/issue to tackle separately, either before or after CD grid?  And why weren't dxt and dyt based on HUS and HUW in prior implementations?

Reply:
Okay.  We are identifying lots of issues to look into later.  So POP doesn't use HUS and HUW either?  This goes WAY back, farther than my memory serves, but for the infrastructure stuff I copied POP as closely as possible, and there was an effort back then to reduce memory usage wherever we could, so maybe we chose to not read them in.  dxt and dyt are calculated from HTN and HTE (e.g. dxt=0.5*(HTNij+HTNij-1)).  dxu and dyu would be averaged similarly from HUS and HUW.  I'm fine with continuing to do the analogous calculations for now, recognizing that there are some sanity checks we could do using the originally calculated grid lengths from the files.

Reply:
I looked at POP.  It computes the dx, dy just like CICE currently.  But it also reads HUS, HUW and stores it internally and uses it for a few computations not directly related to grid lengths.  Very odd.  I propose we add a namelist variable that allows us to switch back and forth between "old" and "new" grid computations.